### PR TITLE
Feature/savings log

### LINKED ISSUE #27: Create Model for SavingLog
A SavingLog is used to record an instance where a PurchaseItem is ticked off and the price in the ExpensLog  lower than that of the PurchaseItem or alternatively when the total spent in a section is less than the total of PurchaseItems in the section, it will contain the following fields 

- name: name of the SavingLog 
- amount: amount saved 
- log_category_id: name of the LogCategory this saving is attributed to 
- money_log _id: id of the MoneyLog this SavingsLog belongs to

a model will be needed to 
- [ ] Create a savings log 
- [ ] Update a savings log

### PR DESCRIPTION
**Pull request by r-coh**
Fixes #27

## Proposed Changes

  -create savings log
  -update savings log
  -

### CODE REVIEW

**Comment by zacck-zz** [comment]
@br4inii  it seems like the code is broken please run `MIX_ENV=test mix test`

File: priv/repo/migrations/20180716142600_create_savings_log.exs [hunk 1]
```diff
@@ -0,0 +1,20 @@
+defmodule Spender.Repo.Migrations.SavingsLog do
+  @moduledoc """
+  This module holds the database schema for the SavingsLog tables,
+  """
+  use Ecto.Migration
+
+  def change do
+    @doc """
```
Comment (zacck-zz) [inline priv/repo/migrations/20180716142600_create_savings_log.exs]: hey @br4inii  this doc has to be outside the function

File: priv/repo/migrations/20180716142600_create_savings_log.exs [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    create a table with SavingsLog details
+    """
+    create table(:savingslog) do
+      add :name, :string
+      add :amount, :float
+      add :log_category_id, references(:budgets)
```
Comment (zacck-zz) [inline priv/repo/migrations/20180716142600_create_savings_log.exs]: this should be referencing `LogCategories`

**Review by zacck-zz — CHANGES_REQUESTED** [review]
@br4inii  good work on this, however, we need a couple of changes to get this going

**Review by zacck-zz — CHANGES_REQUESTED** [review]
@br4inii  this is awesome excellent work looks clean!
Would it be possible for you to squash some of the earlier commits that are not relevant now?

**Comment by r-coh** [comment]
@zacck am on it!!

**Review by zacck-zz — APPROVED** [review]
@br4inii  this is awesome work!

**Pull request merged**